core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/DeclarativeShadowVariableDescriptor.java

```diff
@@ -80,6 +83,19 @@ A shadow variable must have at least one source (since otherwise it a constant).
                     Maybe add one source?
                     """.formatted(methodName, ShadowVariable.class.getSimpleName(), variableMemberAccessor));
         }
+
+        if (shadowVariableUpdater.groupKey() != null && !shadowVariableUpdater.groupKey().isEmpty()) {
+            var groupKey = shadowVariableUpdater.groupKey();
+            Member member = RootVariableSource.getMember(entityDescriptor.getEntityClass(),
+                    groupKey, entityDescriptor.getEntityClass(),
+                    groupKey);
+            groupKeyMap =
+                    entityDescriptor.getSolutionDescriptor().getMemberAccessorFactory().buildAndCacheMemberAccessor(member,
+                            MemberAccessorFactory.MemberAccessorType.FIELD_OR_GETTER_METHOD, ShadowSources.class,
+                            descriptorPolicy.getDomainAccessType())::executeGetter;
```

Comment:
This member accessor loads a map? How?
The entity should have a field with the group key, I don't understand how it suddenly reads a map, and where from.